metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -273,14 +324,267 @@ void main() {
     );
 
     test(
-      ".fetchBuildsWithStatus() returns an empty list",
+      ".fetchBuildsWithStatus() throws an ArgumentError if the given project id is null",
+      () {
+        expect(
+          () => adapter.fetchBuildsWithStatus(null, BuildStatus.successful),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws an ArgumentError if the given build status is null",
+      () {
+        expect(
+          () => adapter.fetchBuildsWithStatus(testProjectId, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws a DestinationError if fetching a project with the given project id fails",
```

Comment:
Let's ensure here or in another test case that if a project does not exist then the method doesn't try to fetch builds.